### PR TITLE
Clean-up, Refactoring, and Fixes

### LINKED ISSUE #816: [1.7.10][RTG-1.0.1] Crashes on creating non-RTG worlds
Hello.
I found simple and strange bug. If I understand correctly, RTG tries to properly select material for villages in desert biomes.
But there is bug, that crashes the world, if world type **is not** _Realistic_. (I just tried to create simple superflat world).
Also, as I see in code, RTG doesn't handle villages in desert biomes if UBC is installed. (UBC has config option to replace all stone stuff in villages). And I often see  (_UBfied_) cobblestone villages in desert.
Please, fix these problems (check type of world and check UBC stones in villages). Both these ones are in [this method](https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/1.7.10-master/src/main/java/rtg/event/EventManagerRTG.java#L189).

[Crash report](http://pastebin.com/b1vU8hFM).

P. S. Sorry for my English, I'm from Russia.

### PR DESCRIPTION
**Pull request by srs-bsns**
Changed EventManager registration, closes #816
Refactored and simplied the config manager
Removed various unneeded WorldType checks
Other code cleanup as per Team-RTG code style

### CODE REVIEW

**Comment by srs-bsns** [comment]
@Team-RTG/everyone Everything should be good with this, but you might want to check it yourself.

File: src/main/java/rtg/world/WorldTypeRTG.java [hunk 1]
```diff
@@ -80,10 +53,30 @@ public IChunkProvider getChunkGenerator(World world, String generatorOptions)
...
+
+    private static Runnable clearChunkProvider() {
+        return new Runnable() {
+            public void run() {
+                chunkProvider = null;
+            }
+        };
+    }
+
+    public boolean isWorldTypeRTG() {
```
Comment (topisani) [inline src/main/java/rtg/world/WorldTypeRTG.java]: you cant do that, it has a high chance of screwing up on servers with multiple worlds

Comment (srs-bsns) [inline src/main/java/rtg/world/WorldTypeRTG.java]: The Runnable methods? I never added them, they were [added by @Zeno410](https://github.com/Team-RTG/Realistic-Terrain-Generation/commit/7704310c85590959fb4eef6879c19a879e66e92f), like 2 months ago. I just moved the methods in the file. (Well, automatic code clean-up did.)

The Runnable methods don't do anything until the server is stopped, at which time they reset the RTG Chunk Manager/Provider.

I did, however, add isWorldTypeRTG(). What's wrong with that?

Comment (topisani) [inline src/main/java/rtg/world/WorldTypeRTG.java]: that wass the one i was refering to. It may screw up when the server has more than one world

**Pull request closed without merging**